docs/source/dataframe-parquet.rst

```diff
@@ -0,0 +1,234 @@
+    ...     columns=["a", "b", "c"]  # Only read columns 'a', 'b', and 'c'
+    ... )
+
+
+Writing
+-------
+
+.. autosummary::
+    to_parquet
+    DataFrame.to_parquet
+
+Dask dataframe provides a `to_parquet` function (as well as a method) for
+writing parquet files and datasets.
+
+In its simplest usage, this takes a path to the directory in which to write the
+dataset. Just as with `read_parquet`, this path may be local, or point to some
+remote filesystem (e.g. S3, HDFS, ...) by prepending the path with a protocol.
+
+.. code-block:: python
+
+    # Write to a local directory
+    >>> df.to_parquet("path/to/my/parquet/")
+
+    # Write to S3
+    >>> df.to_parquet("s3://bucket-name/my/parquet/")
+
+Note that for remote filesystems you many need to configure credentials. When
+possible we recommend handling these external to Dask through
+filesystem-specific configuration files/environment variables (e.g. the
+``~/.aws/config`` file for S3). Alternatively, you can pass configuration on to
+the fsspec_ backend through the ``storage_options`` kwarg:
+
+.. code-block:: python
+
+   >>> df.to_parquet(
+   ...     "s3://bucket-name/my/parquet/",
+   ...     storage_options={'anon': True}  # passed to `s3fs.S3FileSystem`
+   ... )
+
+Dask will write one file per dask dataframe partition to this directory. To
+optimize access for downstream consumers, we recommend aiming for an in-memory
+size of 10-250 MiB per partition. This helps balance worker memory usage
+against dask overhead. You may find the `DataFrame.memory_usage_per_partition`
+method useful for determining if your data is partitioned optimally.
+
+`to_parquet` has many configuration options affecting both behavior and
+performance. Here we highlight a few common options.
+
+Engine
+~~~~~~
+
+`to_parquet` supports two backend engines - ``fastparquet`` and ``pyarrow``.
+For historical reasons this defaults to ``fastparquet`` (if installed), and
+falls back to ``pyarrow``. We recommend using ``pyarrow`` when possible. This
+can be explicitly set by passing ``engine="pyarrow"``, but if ``fastparquet``
+is not installed this configuration is unnecessary since it will automatically
+fallback to ``pyarrow``.
+
+.. code-block:: python
+
+   >>> df.to_parquet(
+   ...     "s3://bucket-name/my/parquet/",
+   ...     engine="pyarrow"  # explicitly specify the pyarrow engine
+   ... )
+
+Metadata
+~~~~~~~~
+
+By default dask will write a ``_metadata`` file aggregating row-group metadata
+from all files together. While potentially useful when reading data later with
+Dask, for large datasets the generation of this file may result in excessive
+memory usage (and potentially killed dask workers). As such, we generally
+recommend disabling creation of this file by passing in
+``write_metadata_file=False``.
+
+.. code-block:: python
+
+   >>> df.to_parquet(
+   ...     "s3://bucket-name/my/parquet/",
+   ...     ignore_metadata_file=False  # disable writing the metadata file
```

Comment:
The default is still `True`; we can debase separately whether this is the best default.

Reply:
Agreed, the purpose here was to document our recommendations and some common options, ideally the number of caveats in this document decreases over time as we sand off rough edges.

https://github.com/dask/dask/issues/8901